cloudformation/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,554 @@
+          - ',{"type":"text","x":{{ coord.x }},"y":{{ coord.y }},"width":24,"height":1,"properties":{"markdown":"\n##
+            EFS Metrics\n"}}'
+
+            {#- Unconditional EFS metrics #}
+            {%- set efs_metrics = [{'metrics': ["BurstCreditBalance"], 'extra_params': ['"title":"Burst Credit Balance"']},
+                                   {'metrics': ["ClientConnections"], 'extra_params': ['"title":"Client Connections"']},
+                                   {'metrics': ["TotalIOBytes"], 'extra_params': ['"title":"Total IO Bytes"']},
+                                   {'metrics': ["PermittedThroughput"], 'extra_params': ['"title":"Permitted Throughput"']},
+                                   {'metrics': ["DataReadIOBytes", "DataWriteIOBytes"], 'extra_params': ['"title":"Data Read/Write IO Bytes"']}]
+            -%}
+            {{ update_coord_after_section(1) }}
+            {%- for metrics_param in efs_metrics %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metrics_param.extra_params|join(',') }}{% if metrics_param.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+              {%- for metric in metrics_param.metrics %}
+          - !Sub '{% if not loop.first %},{% else %}[{% endif %}["AWS/EFS","{{ metric }}","FileSystemId","${EFSFileSystemId}"]{{ ']}}' if loop.last else '' }}'
+              {%- endfor -%}
+              {{ update_coord(graph_width, graph_height) }}
+            {%- endfor %}
+
+            {#- Conditional EFS metrics #}
+            {%- set efs_metrics_conditions_params = [{'metric': 'PercentIOLimit', 'supported_vol_types': ["generalPurpose"], 'extra_params': ['"title":"Percent IO Limit"']}] %}
+            {%- set efs_volume_type = cfn_params.EFSOptions.split(',')[2] %}
+            {%- for metric_condition_params in efs_metrics_conditions_params %}
+              {%- if efs_volume_type in metric_condition_params.supported_vol_types %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metric_condition_params.extra_params|join(',') }}{% if metric_condition_params.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+          - !Sub '[["AWS/EFS","{{ metric_condition_params.metric }}","FileSystemId","${EFSFileSystemId}"]]}}'
+              {%- endif %}
+            {%- endfor -%}
+            {{ update_coord(graph_width, graph_height) }}
+
+          {%- endif -%} {#- if efs_shared_dir not in ["NONE", "/NONE"] -#}
+          {{ update_coord_after_section(graph_height) }}
+
+          {%- set fsx_shared_dir = cfn_params.FSXOptions.split(',')[0] %}
+          {%- if fsx_shared_dir not in ["NONE", "/NONE"] %}
+
+          {#- FSx metrics graphs #}
```

Comment:
Same here, let's move this comment before the `set fsx_shared_dir`

Reply:
Done